app/FirebaseApi.js

```diff
@@ -0,0 +1,46 @@
+/* global someFunction window:true */
+
+import firebase from 'firebase';
+import RNFetchBlob from 'react-native-fetch-blob';
+import Config from 'react-native-config';
+
+// Init Firebase
+const config = {
+  apiKey: Config.FIREBASE_API_KEY,
+  authDomain: 'freemarketrn.firebaseapp.com',
+  storageBucket: 'freemarketrn.appspot.com',
+};
+firebase.initializeApp(config);
+const storage = firebase.storage();
+
+// Prepare Blob support
+const Blob = RNFetchBlob.polyfill.Blob;
+const fs = RNFetchBlob.fs;
+window.XMLHttpRequest = RNFetchBlob.polyfill.XMLHttpRequest;
+window.Blob = Blob;
+
+export const saveImageToStorage = localImage => (
+  new Promise((resolve, reject) => {
+    const uploadUri = localImage.uri;
+    const mime = 'application/octet-stream';
+    const sessionId = new Date().getTime();
+    let uploadBlob = null;
+    const imageRef = storage.ref('images').child(`${sessionId}${localImage.fileName}`);
+    fs.readFile(uploadUri, 'base64')
+      .then(data => Blob.build(data, { type: `${mime};BASE64` }))
```

Comment:
And don't forget that we are working in well established development environment, no need to use web hacks for such simple task as file reading. If we have to, then ReactNative is just unusable and we should consider wrap native APIs in component or such.